build/scripts/script.js

```diff
@@ -519,84 +520,146 @@ function updateSearchResults(value) {
-        linkH1.href = href;
-      } else {
-        linkH1 = document.createElement("span");
+  for (const file of Object.values(searchResultSorted)) {
+    for (const h1 of Object.values(file)) {
+      if (h1.__DEFAULT__) {
+        const elem = createResultElement(h1.__DEFAULT__);
+        searchResultsElt.appendChild(elem);
       }
-      linkH1.className = "h1";
-      linkH1.textContent = res.item.h1;
-      locationDiv.appendChild(linkH1);
-      needSeparator = true;
-    }
+      for (const [keyH2, valueH2] of Object.entries(h1)) {
+        if (keyH2 === "__DEFAULT__") {
+          continue;
+        }
+        if (valueH2.__DEFAULT__) {
+          const elem = createResultElement(valueH2.__DEFAULT__);
+          searchResultsElt.appendChild(elem);
+        }
 
-    if (res.item.h2 !== undefined && res.item.h2 !== "") {
-      if (needSeparator) {
-        const separatorSpan = document.createElement("span");
-        separatorSpan.textContent = " > ";
-        locationDiv.appendChild(separatorSpan);
-        needSeparator = false;
-      }
-      let linkH2;
-      if (links.anchorH2 !== undefined) {
-        const href = rootUrl + "/" + links.file + "#" + links.anchorH2;
-        linkH2 = document.createElement("a");
-        linkH2.href = href;
-      } else {
-        linkH2 = document.createElement("span");
+        for (const [keyH3, valueH3] of Object.entries(valueH2)) {
+          if (keyH3 === "__DEFAULT__") {
+            continue;
+          }
+          valueH3.forEach((val) => {
+            const elem = createResultElement(val);
+            searchResultsElt.appendChild(elem);
+          });
+        }
       }
-      linkH2.className = "h2";
-      linkH2.textContent = res.item.h2;
-      locationDiv.appendChild(linkH2);
-      needSeparator = true;
     }
-    if (res.item.h3 !== undefined && res.item.h3 !== "") {
-      if (needSeparator) {
-        const separatorSpan = document.createElement("span");
-        separatorSpan.textContent = " > ";
-        locationDiv.appendChild(separatorSpan);
-        needSeparator = false;
+  }
+}
+
+/**
+ * Add missing top sections headers (h1) in search results.
+ * Search result can be h1, h2 or h3, but the search algorithm can
+ * find a match for a h3 section and not for the h1.
+ * To not display h3 orphan in the tree view, we manually add to
+ * the search result the h1 that is a parent to that h3.
+ * @param {array} searchResults
+ * @returns
+ */
+function addMissingTopSections(searchResults) {
+  const h1Sections = {};
+  const missingH1 = [];
+  for (result of searchResults) {
+    if (!h1Sections[result.item.h1]) {
+      h1Sections[result.item.h1] = [];
+    }
+    h1Sections[result.item.h1].push(result);
+  }
+
+  for (const value of Object.values(h1Sections)) {
+    const topSection = value.find((r) => {
+      return r.item.h2 === undefined;
+    });
+    if (topSection === undefined) {
+      const fakeH1 = structuredClone(value[0]);
```

Comment:
I've no idea what we're trying to do there with those `fakeH1 = ...` (fakeH1 is an object?), it may need comments